luigi/contrib/dropbox.py

```diff
@@ -0,0 +1,275 @@
+                                                   ntpath.basename(path))
+        self.fid = None
+        self.closed = False
+
+    def read(self):
+        return self.client.download_as_bytes(self.path)
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc, traceback):
+        self.close()
+
+    def __del__(self):
+        self.close()
+        if os.path.exists(self.download_file_location):
+            os.remove(self.download_file_location)
+
+    def close(self):
+        self.closed = True
+
+    def readable(self):
+        return True
+
+    def writable(self):
+        return False
+
+    def seekable(self):
+        return False
+
+    def seek(self, offset, whence=None):
+        pass
+
+
+class AtomicWritableDropboxFile(AtomicLocalFile):
+    def __init__(self, path, client):
+        """
+        Represents a file that will be created inside the Dropbox cloud
+
+        :param str path: Destination path inside Dropbox
+        :param DropboxClient client: a DropboxClient object (initialized with a valid token, for the desired account)
+        """
+        super(AtomicWritableDropboxFile, self).__init__(path)
+        self.path = path
+        self.client = client
+
+    def move_to_final_destination(self):
+        """
+        After editing the file locally, this function uploads it to the Dropbox cloud
+        """
+        self.client.upload(self.tmp_path, self.path)
+
+
+class DropboxTarget(FileSystemTarget):
+    """
+    A Dropbox filesystem target.
+    """
+
+    def __init__(self, path, token=None, format=None):
+        """
+        Create an Dropbox Target for storing data in a dropbox.com account
+
+        **About the path parameter**
+
+        The path must start with '/' and must not end with '/' (even if it is a directory).
+        If the app has 'App folder' access, then / will refer to this app folder (which
+        mean that there is no need to prepend the name of the app to the path)
+        Otherwise, if the app has 'full access', then / will refer to the root of the Dropbox folder
+
+
+        **About the token parameter:**
+
+        The Dropbox target requires a valid OAuth2 token as a parameter (which means that a `Dropbox API app
+        <https://www.dropbox.com/developers/apps>`_ must be created. This app can have 'App folder' access
+        or 'Full Dropbox', as desired).
+
+        Information about generating the token can be read here:
+
+        - https://dropbox-sdk-python.readthedocs.io/en/latest/api/oauth.html#dropbox.oauth.DropboxOAuth2Flow
+        - https://blogs.dropbox.com/developers/2014/05/generate-an-access-token-for-your-own-account/
+
+        :param str path: Remote path in Dropbox (starting with '/').
+        :param str token:
+        :param luigi.Format format: the luigi format to use (e.g. `luigi.format.Nop`)
+
+
+        """
+        super(DropboxTarget, self).__init__(path)
+
+        if not token:
```

Comment:
if token is required, why does it have a `None` kwarg default value?

Reply:
You are right, I have changed it.